src/main/java/com/google/sps/servlets/UpdateVendorServlet.java

```diff
@@ -92,26 +92,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         ? new Entity("Picture", vendorKey)
         : new Entity("Picture", vendorObject.getProfilePic().getId(), vendorKey);
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-
     // If not the same, delete previous blob from blobstore
-    if (vendorObject.getProfilePic() != null && 
+    if (vendorObject.getProfilePic() != null && imageBlobKey != null &&
         imageBlobKey.compareTo(vendorObject.getProfilePic().getBlobKey()) != 0) {
       BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
       blobstoreService.delete(vendorObject.getProfilePic().getBlobKey());
     }
-    picture.setProperty("blobKey", imageBlobKey);
-    picture.setProperty("altText", altText);
-    datastore.put(picture);
 
-    EmbeddedEntity picInfo = new EmbeddedEntity();
-    picInfo.setKey(picture.getKey());
-    picInfo.setPropertiesFrom(picture);
+    // If there is a picture with alt text, set properties; Else if only picture, delete from blobstore 
```

Comment:
Nit: I'd move this comment to the first line of the else if block.

Reply:
For sure!